lib/src/block_syntaxes/footnote_def_syntax.dart

```diff
@@ -0,0 +1,74 @@
+import '../ast.dart' show Element, Node;
+import '../block_parser.dart' show BlockParser;
+import '../line.dart';
+import '../patterns.dart' show footnotePattern;
+import 'block_syntax.dart' show BlockSyntax;
+import 'empty_block_syntax.dart';
+import 'paragraph_syntax.dart';
+import 'setext_header_syntax.dart';
+
+class FootnoteDefSyntax extends BlockSyntax {
```

Comment:
Would you mind adding a doc comment explaining the source of this syntax? Thanks!

Reply:
This needs to be addressed, please.

Reply:
I'm sorry, @lindeer , we cannot accept this PR unless this comment is addressed. We need to have documented where the syntax is specified. Who controls its definition / where can we go for updates, etc?

Reply:
sorry @srawlins , I have no idea why I missed these comment..

Reply:
I came to know that [footnotes are not in the spec yet](https://github.com/github/cmark-gfm/issues/283#issuecomment-1378868725).  It is said that there is a private repo for the spec somewhere in GH.
I found a GFM footnote grammar and its rust implementation [here](https://github.com/wooorm/markdown-rs/blob/2498e31eecead798efc649502bbf5f86feaa94be/src/construct/gfm_footnote_definition.rs).
or what other I could do to move this PR forward?

Reply:
Sorry, I was on vacation for a few days 😭 but we're really almost there. I'd love to land this with like one more change 😁 . It is totally fine that there is not a published spec; that's all we have so that's ok. I do still love this feature because it helps us match what GitHub parses. So thanks again for your work!

With a feature that is under-specified like this, I think it' just important to add all of the links we can 😁 If you could add, in the doc comment links to:

* the issue: https://github.com/github/cmark-gfm/issues/283#issuecomment-1378868725
* the cmark-gfm source: https://github.com/github/cmark-gfm/blob/master/src/blocks.c#L1212
* the rust implementation is a great resource too, I think you can add it for good measure: https://github.com/wooorm/markdown-rs/blob/2498e31eecead798efc649502bbf5f86feaa94be/src/construct/gfm_footnote_definition.rs